packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -581,6 +581,139 @@ class _UiKitPlatformView extends LeafRenderObjectWidget {
+  final PlatformViewCreatedCallback onPlatformViewCreated;
+}
+
+/// The factory to construct a [PlatformViewSurface].
+///
+/// It is required when constructing a [PlatformViewLink].
+typedef PlatformViewSurfaceFactory = PlatformViewSurface Function(BuildContext context, PlatformViewController controller);
+
+///Constructs a [PlatformViewController].
+///
+/// The [PlatformViewController.id] field of the created controller must match the value of the
+/// params [PlatformViewCreationParams.id] field.
+///
+/// See also [PlatformViewLink.onCreate].
+typedef CreatePlatformViewController = PlatformViewController Function(PlatformViewCreationParams params);
+
+/// Links a platform view with the Flutter framework.
+///
+/// Provides common functionality for embedding a platform view (e.g an android.view.View on Android)
+/// with the Flutter framework.
+///
+/// {@macro flutter.widgets.platformViews.lifetime}
+///
+/// To implement a new platform view widget, return this widget in the `build` method.
+/// For example:
+/// ```dart
+/// class FooPlatformView extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return PlatformViewLink(
+///       createCallback: createFooWebView,
+///       surfaceFactory: (BuildContext context, PlatformViewController controller) {
+///        return PlatformViewSurface(
+///            gestureRecognizers: gestureRecognizers,
+///            controller: controller,
+///            hitTestBehavior: PlatformViewHitTestBehavior.opaque,
+///        );
+///       },
+///    );
+///   }
+/// }
+/// ```
+///
+/// The `surfaceFactory` and the `createPlatformViewController` only take affect when the state of this widget is initialized.
+/// If the widget is rebuilt without losing its state, `surfaceFactory` and `createPlatformViewController` are ignored.
+class PlatformViewLink extends StatefulWidget {
+
+  /// Construct a [PlatformViewLink] widget.
+  ///
+  /// The `surfaceFactory` and the `createPlatformViewController` must not be null.
+  ///
+  /// See also:
+  /// * [PlatformViewSurface] for details on the widget returned by `surfaceFactory`.
+  /// * [PlatformViewCreationParams] for how each parameter can be used when implementing `createPlatformView`.
+  const PlatformViewLink({
+    Key key,
+    @required PlatformViewSurfaceFactory surfaceFactory,
+    @required CreatePlatformViewController createPlatformViewController,
+    }) : assert(surfaceFactory != null),
+                                  assert(createPlatformViewController != null),
+                                  _surfaceFactory = surfaceFactory,
+                                  _createPlatformViewController = createPlatformViewController,
+                                  super(key: key);
+
+
+  final PlatformViewSurfaceFactory _surfaceFactory;
+  final CreatePlatformViewController _createPlatformViewController;
+
+  @override
+  State<StatefulWidget> createState() {
```

Comment:
style nit: use `=>` if it fits in one line.

Reply:
done